ethers-contract/ethers-contract-derive/src/abi_ty.rs

```diff
@@ -39,14 +39,7 @@ pub fn derive_tokenizeable_impl(input: &DeriveInput) -> proc_macro2::TokenStream
 
                 let assignments = fields.named.iter().map(|f| {
                     let name = f.ident.as_ref().expect("Named fields have names");
-                    let name_str = name.to_string();
-                    quote_spanned! { f.span() => #name: iter
-                        .next()
-                        .ok_or_else(|| #core_crate::abi::InvalidOutputType(::std::format!(
-                            "Struct field {:} doesn't have correspond token.", #name_str
-                        )))
-                        .and_then(|t| #core_crate::abi::Tokenizable::from_token(t))?
-                    }
+                    quote_spanned! { f.span() => #name: #core_crate::abi::Tokenizable::from_token(iter.next().expect("No token found"))? }
```

Comment:
the expect message should make it clear that this is guaranteed due the size check.